File: backend/orchestrator-prompt/system/error_analysis.py
def get_error_analysis_message(
    programming_language: str, error_message: str, considerations: str = None
) -> str:
    ERROR_ANALYSIS_SYSTEM_MESSAGE = f"""\
# 役割
あなたは{programming_language}に精通した技術エキスパートです。

# 目的
ユーザから提供されたエラーメッセージの意味と解決策を分かりやすく説明する。

# 指示
以下の"#流れ"に従ってstep by stepでエラーを解析してください。

# 流れ
1. ユーザから受け取った"#エラーメッセージ"を確認する。
2. エラーの種類と原因を特定する。
3. エラーが発生する典型的な状況を把握する。
4. 具体的な解決策を考える。
5. 修正の例があれば提示する。
6. エラーの説明、解決策、修正例をユーザに出力する。

# 制約条件
- "#出力形式"に従うこと。
- 技術的な説明は分かりやすく、初心者にも理解できるレベルで説明すること。
- 解決策は具体的で実践可能なものを提示すること。
- 可能であれば、修正前後のコード例を示すこと。
- 回答は「日本語」で行うこと。
- エラーの背景や関連する概念についても簡潔に説明すること。

# 入力欄
プログラミング言語：{programming_language}
エラー：{error_message}
{f"考慮事項：{considerations}" if considerations else ""}

# 出力形式
## エラーの説明
[エラーの意味と発生原因を分かりやすく説明]

## 解決策と修正例
[具体的な解決方法を段階的に説明し、可能であれば修正前後のコード例を提示]
"""

    return ERROR_ANALYSIS_SYSTEM_MESSAGE

File: frontend/src/app/_store/slice/error-analysis.ts
import { createSlice, PayloadAction } from '@reduxjs/toolkit';

export interface ErrorAnalysisState {
  id: string;
  programmingLanguage: string;
  errorMessage: string;
  considerations: string;
  feedbackAt: undefined | Date;
}

export interface InitialState extends ErrorAnalysisState {
  result: {
    explanation: string;
    solutionAndExample: string;
  };
}

export const initialErrorAnalysis: ErrorAnalysisState = {
  id: '',
  programmingLanguage: '',
  errorMessage: '',
  considerations: '',
  feedbackAt: undefined,
};

export const initialState: InitialState = {
  ...initialErrorAnalysis,
  result: {
    explanation: '',
    solutionAndExample: '',
  },
};

export const errorAnalysisSlice = createSlice({
  name: 'errorAnalysis',
  initialState,
  reducers: {
    setErrorAnalysis: (state, action: PayloadAction<ErrorAnalysisState>) => ({
      ...state,
      ...action.payload,
    }),
    setResult: (
      state,
      action: PayloadAction<{
        result: { explanation: string; solutionAndExample: string };
        feedbackAt: Date | undefined;
      }>
    ) => ({
      ...state,
      result: action.payload.result,
      feedbackAt: action.payload.feedbackAt,
    }),
    setId: (state, action: PayloadAction<string>) => ({
      ...state,
      id: action.payload,
    }),
    setFeedbackAt: (state, action: PayloadAction<Date>) => ({
      ...state,
      feedbackAt: action.payload,
    }),
    setReset: (state) => ({
      ...state,
      ...initialState,
    }),
  },
});

export const { setResult, setReset, setErrorAnalysis, setFeedbackAt, setId } =
  errorAnalysisSlice.actions;
export const errorAnalysisReducer = errorAnalysisSlice.reducer;
